internal/worker/cmd_decompose.go

```diff
@@ -135,3 +142,54 @@ func decomposeMGet(_ context.Context, _ *BaseWorker, cd *cmd.DiceDBCmd) ([]*cmd.
 	}
 	return decomposedCmds, nil
 }
+
+func decomposeSInter(_ context.Context, _ *BaseWorker, cd *cmd.DiceDBCmd) ([]*cmd.DiceDBCmd, error) {
+	if len(cd.Args) < 1 {
+		return nil, diceerrors.ErrWrongArgumentCount("SINTER")
+	}
+	decomposedCmds := make([]*cmd.DiceDBCmd, 0, len(cd.Args))
+	for i := 0; i < len(cd.Args); i++ {
+		decomposedCmds = append(decomposedCmds,
+			&cmd.DiceDBCmd{
+				Cmd:  store.Smembers,
+				Args: []string{cd.Args[i]},
+			},
+		)
+	}
+	return decomposedCmds, nil
+}
```

Comment:
Note for future ref: I have thoughts about how this code could be reused. For now, we can let it be, but we can note that since more commands from the set data type will have the same decompose, we might be able to reuse this logic.

Reply:
Yes, Once commands are migrated we will reuse the logic being replicated in multiple compose and decompose. Thanks for mentioning this, already have this in plan.